bql/semantic/expression_test.go

```diff
@@ -412,11 +430,76 @@ func TestNewEvaluator(t *testing.T) {
 				}),
 			},
 			r: table.Row{
-				"?foo": &table.Cell{S: table.CellString(buildLiteral("\"abc\"^^type:text"))},
+				"?foo": &table.Cell{
+					L: buildLiteralOrDie(`"abc"^^type:text`),
+				},
+			},
+			err:  false,
+			want: true,
+		},
+		{
+			id: "?s ID ?id = \"abc\"^^type:text",
+			in: []ConsumedElement{
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemBinding,
+					Text: "?id",
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemEQ,
+				}),
+				NewConsumedToken(&lexer.Token{
+					Type: lexer.ItemLiteral,
+					Text: "\"abc\"^^type:text",
+				}),
+			},
+			r: table.Row{
+				"?id": &table.Cell{S: table.CellString("abc")},
+			},
+			err:  false,
+			want: true,
+		},
+		{
+			id: "?s ID ?id < \"bbb\"^^type:text",
```

Comment:
If the escaped quotes starts to bother you, it is possible to also write this as:
`?s ID ?id < "bbb"^^type:text`

Reply:
Cool, I did not know about that. Thank you Kloss.